IHP/IDE/CodeGen/Controller.hs

```diff
@@ -27,8 +27,16 @@ instance Controller CodeGenController where
 
     action NewControllerAction = do
         let controllerName = paramOrDefault "" "name"
-        plan <- ControllerGenerator.buildPlan controllerName
-        render NewControllerView { .. }
+        controllerAlreadyExists <- case Text.splitOn "." controllerName of
+            [applicationName, controllerName'] -> doesFileExist $ cs applicationName <> "/Controller/" <> cs controllerName' <> ".hs"
+            [controllerName'] -> doesFileExist $ "Web/Controller/" <> cs controllerName' <> ".hs"
```

Comment:
Can we move this into a helper function `doesControllerExists name` (at the bottom of the controller file)? Then we could do just `controllerAlreadyExists <- doesControllerExists name`.

Also right now we have duplicated the controller name handling stuff. Can we have a function `normalizeControllerName :: Text -> (Text, Text)`, which returns `("Admin", "Users")` for e.g. `"admin.users"`. Then this function can be used by this controller and also by the `ControllerGenerator`

Reply:
I think a complete separate function would be of no use because it is and only will be used inside of this action. For better readability we could put it in the where clause.

Reply:
Yeah this is omly for readability :)

Reply:
Done :)